pandas/core/arrays/base.py

```diff
@@ -477,6 +486,16 @@ def take(self, indexer, allow_fill=True, fill_value=None):
            def take(self, indexer, allow_fill=True, fill_value=None):
                indexer = np.asarray(indexer)
                mask = indexer == -1
+
+               # take on empty array not handled as desired by numpy
+               if not len(self):
+                   # only valid if result is an all-missing array
+                   if mask.all():
+                       return type(self)([self._na_value] * len(indexer))
+                   else:
+                       raise IndexError(
+                           "cannot do a non-empty take from an empty array.")
```

Comment:
@TomAugspurger Adding the full example is maybe a bit too much (it defeats a bit the purpose of the small illustrative example)? On the other hand, it is needed for a correct implementation.

Reply:
Agreed... I think that just having a comment noting the correct thing to do for `.take` on an empty EA is fine.

Reply:
@TomAugspurger I added some explanation in the docstring, but I decided to keep the code example as well: it is now a bit shorter + it is actually really needed to get a correct implementation.